### ISSUE TITLE
Add support for MariaDB's "system-versioned tables" (temporal tables)

### ISSUE THREAD

**Issue by lauxjpn** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1519]
The SQL Server provider recently added support for [temporal tables](https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-6.0/whatsnew#sql-server-temporal-tables).

While MySQL has not yet added support for temporal tables, MariaDB has in form of their _system-versioned tables_.

We should consider supporting those.

**Comment by woojunfeng** [comment]
any progress?

**Comment by lauxjpn** [comment]
@woojunfeng No, but since the community seems to want this, we will consider it for 7.0.

**Comment by kryptobi** [comment]
Do you think this will be released soon after the release of efcore 7?

**Comment by Hoopou** [comment]
> @woojunfeng No, but since the community seems to want this, we will consider it for 7.0.

Any update on this?

**Comment by lauxjpn** [comment]
We haven't done any work on this yet. But as there seems to be general interest in that feature, we should consider it for 8.0.

**Comment by kryptobi** [comment]
> We haven't done any work on this yet. But as there seems to be general interest in that feature, we should consider it for 8.0.

That would be a nice!

**Comment by kryptobi** [comment]
Any progress?

**Comment by okuhlen** [comment]
> We haven't done any work on this yet. But as there seems to be general interest in that feature, we should consider it for 8.0.

Thank you!

**Comment by yhdezalvarez** [comment]
Any progress?

**Comment by roji** [comment]
@lauxjpn it may be interesting to explore any common ground between MariaDB's feature and the SQL Server temporal tables feature that's already available - maybe let's keep in touch if/when you get around to working on this...

/cc @maumar

**Comment by roji** [comment]
OK! Hopefully there's some opportunity for even sharing some code (possibly after duplicating into your provider)? Maybe keep it in mind as you're working...

(also it may be time for that yearly beer ;))

**Comment by yhdezalvarez** [comment]
Any updates on this issue?

**Comment by manuelesvi** [comment]
I'd love to see this feature implemented for MariaDB. I can help testing it, if there's any progress on it.

**Comment by okuhlen** [comment]
@roji : I am wondering if leveraging System-Versioned tables for MariaDB would not mean using as possible from the code in the `SqlServerMigrationsSqlGenerator.cs`. in the EFCore.SqlServer project, in the EFCore repository. It looks interesting - from glossing over the code, this makes me think it could be possible to reuse as much as possible here. I don't know the full details but my hunch is saying that. I kinda do need this functionality on a personal project I am working on, so I wonder if there is no way to re-use and re-purpose this code for MariaDB - I guess the complexity is that this is only on MariaDB; not sure about MySQL. 

![Image](https://github.com/user-attachments/assets/dad8dd7f-bfb6-4527-9f9f-d6942aaa17b1)

**Comment by okuhlen** [comment]
@roji Thanks a lot for the response! Hmm, lemme see if I can use the little I know about MariaDB for my project. One won't really know until they "get their hands dirty" :)

**Comment by kryptobi** [comment]
@okuhlen if u got something to review or test, let me know. 
I'm also very interested in this topic. Perhaps I can help.

**Comment by kryptobi** [comment]
something new here?..
